### PR TITLE
chore(a11y): Improve granularity of `DateInput` errors

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Addresses issue `DAC_Vague_Error_Message_Usability_01` in Accessibility Report (page 73)
 - Adds granular error messages for missing day, month, or year values
 - Adds granular error messages for invalid day, month, or year values
 - Adds tests for the above errors
 - Splits existing DateInput tests into `Public.test.ts` (react-testing-library) and `model.test.ts` (jest)

https://github.com/theopensystemslab/planx-new/assets/20502206/e146bd53-59b2-4777-a60d-ac8be69ec68e

### CODE REVIEW

**Comment by DafyddLlyr** [comment]
[Vague error message [Usability]](https://trello.com/c/gjnZxPLM/2827-vague-error-message-usability)

File: editor.planx.uk/src/@planx/components/DateInput/model.test.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+  // Leaves valid dates alone
+  expect(paddedDate("2021-01-06", "blur")).toBe("2021-01-06");
+  expect(paddedDate("2021-04-22", "blur")).toBe("2021-04-22");
+  expect(paddedDate("2021-08-04", "blur")).toBe("2021-08-04");
+
+  // Leaves single 0 alone
+  expect(paddedDate("2021-0-2", "blur")).toBe("2021-0-02");
+  expect(paddedDate("2021-10-0", "blur")).toBe("2021-10-0");
+});
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/DateInput/model.test.ts:148]: Helpful split :+1:

File: editor.planx.uk/src/@planx/components/DateInput/model.ts [hunk 2]
```diff
@@ -64,41 +59,81 @@ const displayDate = (date: string): string | undefined => {
...
-      // test runs regardless of required status, so don't fail it if it's undefined
-      return Boolean(!date || isDateValid(date));
-    },
-  );
+  return string()
+    .test(
+      "missing day",
+      "Date must include a day",
+      (date?: string) => {
+        console.log({date})
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/DateInput/model.ts:73]: nit: please clean up this `console.log()` !

**Review by jessicamcinchak — APPROVED** [review]
Smart ! :calendar:

**Pull request merged**